### PR TITLE
Update pip rules to support native bdists and Windows deployment

### PR DESCRIPTION
**Pull request by dmikhalin**
## What is the goal of this PR?

We modify deployment script in order to generate platform-specific pip packages for different python versions. 

## What are the changes implemented in this PR?

We added:
- `"suffix"` attribute to `assemble-pip` rule
- `"suffix"` and `"distribution_tag"` attributes to `deploy-pip` rule

These attributes are used to construct correct filenames of pip packages.

### CODE REVIEW

File: pip/rules.bzl [hunk 1]
```diff
@@ -317,6 +326,14 @@ deploy_pip = rule(
             default = "",
             doc = "Repository name in the .pypirc profile to deploy to"
         ),
+        "distribution_tag": attr.string(
```
Comment (flyingsilverfin) [inline pip/rules.bzl:329]: Let's check what fails if we use an invalid tag - if it fails in PIP that's fine

File: pip/assemble.py [hunk 2]
```diff
@@ -135,4 +164,9 @@ def split_path(path: str) -> list[str]:
 
 shutil.copy(sdist_archives[0], args.output_sdist)
 shutil.copy(wheel_archives[0], args.output_wheel)
-shutil.rmtree(pkg_dir)
+
+# Ignore permission errors (for Windows)
+try:
+    shutil.rmtree(pkg_dir)
+except PermissionError:
+    pass
```
Comment (flyingsilverfin) [inline pip/assemble.py:172]: Print out the error as a warning

File: pip/requirements.txt [hunk 3]
```diff
@@ -20,6 +20,7 @@ pkginfo==1.9.6
 pycparser==2.21
 pygments==2.14.0
 pyparsing==3.0.9
+pywin32-ctypes==0.2.2; sys_platform == 'win32'
```
Comment (flyingsilverfin) [inline pip/requirements.txt:23]: Do we need this

File: pip/templates/deploy.py [hunk 4]
```diff
@@ -88,6 +94,13 @@ def upload_command(repo_type_key, package_file, wheel_file):
...
+    # Do not upload a package file if we build for specific platform, because we use a precompiled library
+    if "{distribution_tag}".endswith("any"):
+        twine.commands.upload.main(upload_command(repo_type_key, new_package_file, new_wheel_pep491))
+    else:
+        twine.commands.upload.main(upload_command(repo_type_key, None, new_wheel_pep491))
 finally:
-    shutil.rmtree(dist_dir)
+    try:
+        shutil.rmtree(dist_dir)
+    except PermissionError:
```
Comment (flyingsilverfin) [inline pip/templates/deploy.py:105]: Print warning

File: pip/templates/deploy.py [hunk 5]
```diff
@@ -76,8 +82,8 @@ def upload_command(repo_type_key, package_file, wheel_file):
 with open("{version_file}") as version_file:
     version = version_file.read().strip()
 try:
-    new_package_file = dist_dir + "/{package_file}".replace(".tar.gz", "-{}.tar.gz".format(version))
-    new_wheel_pep491 = dist_dir + "/{wheel_file}".replace("-", "_").replace(".whl", "-{}-py3-none-any.whl".format(version))
+    new_package_file = dist_dir + "/{package_file}".replace("{suffix}", "").replace(".tar.gz", "-{}.tar.gz".format(version))
+    new_wheel_pep491 = dist_dir + "/{wheel_file}".replace("{suffix}", "").replace("-", "_").replace(".whl", "-{}-{distribution_tag}.whl".format(version))
```
Comment (flyingsilverfin) [inline pip/templates/deploy.py:86]: Let's rename these to `source_package` and `wheel_package`


Let's rename `package_file` to `source_package_name` and `wheel_file` to `wheel_package_name`

File: pip/requirements.txt (hunk 3 above)
Comment (dmitrii-ubskii) [reply pip/requirements.txt:23]: Yes, this is required by keyring on windows and is not fetched automatically.

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**